### ISSUE TITLE
Additional space after the encoded unicode character destroys the meaning of the original text

### ISSUE THREAD

**Issue by jens-duttke** [mathiasbynens/cssesc#20]
The module adds a space after each unicode character, even in your example:
```
cssesc('Ich ♥ Bücher');
// → 'Ich \\2665  B\\FC cher'
```
So, in the result `Bücher` will be shown as `Bü cher` on the website.

Why don't you follow the CSS specification (https://www.w3.org/TR/CSS2/syndata.html#characters) and convert `ü` to `\0000FC`?

To make the result as short as possible is a nice idea, but it doesn't make sense to destroy the orignal string so that the result is not usable anymore.

**Comment by mathiasbynens** [comment]
> So, in the result `Bücher` will be shown as `Bü cher` on the website.

What makes you say that?

**Comment by mathiasbynens** [comment]
It sounds like you're misunderstanding how CSS escape sequences work. The space following the escape sequence is consumed as part of it. Please read <https://mathiasbynens.be/notes/css-escapes>, or [the CSS spec](https://drafts.csswg.org/css-syntax-3/#escaping).

Example: https://mothereff.in/css-escapes#1Ich%20%E2%99%A5%20B%C3%BCcher

Navigate to the following URL to see a demo:

```
data:text/html;charset=utf-8,<!DOCTYPE%20html><title>CSS%20escapes%20example<%2Ftitle><style>pre%7Bbackground%3A%23eee%3Bpadding%3A.5em%7D.test%7Bdisplay%3Anone%7D%23Ich%5C%20%5C2665%20%5C%20B%5CFC%20cher%7Bdisplay%3Ablock%7D.pass%7Bbackground%3Alime%7D.fail%7Bbackground%3Ared%7D<%2Fstyle><h1><a%20href%3D"https%3A%2F%2Fmothereff.in%2Fcss-escapes%231Ich%2520%25E2%2599%25A5%2520B%25C3%25BCcher">Mothereffing%20CSS%20escapes<%2Fa>%20example<%2Fh1><pre><code>Ich%20♥%20Bücher<%2Fcode><%2Fpre><p%20id%3D"Ich%20♥%20Bücher"%20class%3Dtest>If%20you%20can%20read%20this%2C%20the%20escaped%20CSS%20selector%20worked.%20<%2Fp><script>var%20el%3Ddocument.getElementsByTagName('p')%5B0%5D%3Btry%7Bdocument.getElementById('Ich%20%5Cu2665%20B%5CxFCcher').innerHTML%20%2B%3D%20'%20<code>document.getElementById<%2Fcode>%20worked.'%3Bdocument.querySelector('%23Ich%5C%5C%20%5C%5C2665%20%5C%5C%20B%5C%5CFC%20cher').innerHTML%2B%3D'%20<code>document.querySelector<%2Fcode>%20worked.'%3Bel.className%3D'pass'%7Dcatch(e)%7Bel.innerHTML%3D'FAIL'%3Bel.className%3D'fail'%7D<%2Fscript>
```

**Comment by jens-duttke** [comment]
Thanks for quick response and sorry for not going deeper into the details in my initial bug report.

I'm using Webpack with a couple of loaders (sass-loader, postcss-loader, style-loader) and I try to set the content of a pseudo-element like:
```
div::before {
	content: "nächste Seite";
}
```

The output on the website is `nä chste Seite`.

So I did a research and figured out that your module is used to decode UTF-8 characters.

To test if that's really the source of the problem, I've changed 'node_modules/cssesc/cssesc.js' to:
```
	var cssesc = function(string, options) {
		return 'n\\0000E4chste Seite';
	}
```
And everything was fine on my website - that's why I thought your module is the source of the problem.

But it looks like there is another code in any of the loaders which encodes the decoded UTF-8 characters (without to consider the space), and after that decode it again.
The result is, that in the final CSS there are two spaces.

Now I only have to figure out which other code is involved in this problem and place a bug report there ...
